toolkit/pallets/block-production-log/src/test.rs

```diff
@@ -0,0 +1,39 @@
+use super::*;
+use frame_support::traits::UnfilteredDispatchable;
+use mock::*;
+use sp_consensus_slots::Slot;
+use sp_runtime::BoundedVec;
+
+#[test]
+fn first_append_should_succeed() {
+	new_test_ext().execute_with(|| {
+		let call = Call::<Test>::append { id: make_id::<Test>(1) };
+		call.dispatch_bypass_filter(RuntimeOrigin::none()).unwrap();
+
+		assert_eq!(Log::<Test>::get().to_vec(), vec![(Slot::from(1001000), make_id::<Test>(1))]);
+	})
+}
+
+#[test]
+fn append_to_end_of_log() {
```

Comment:
maybe we can merge both testcases, what about following scenario:
1) we test if vec is empty
2) we append something
3) we check if there is elem
4) we append again something else
5) we check if there are both

Reply:
Appending twice is a negative test case. Since we have constant current_slot, it is illegal to run it twice.

Reply:
hmm, but what if we will advance blocks in the mean time, so there will be new session?

Reply:
Slots are monotonic. They are not going to 0 at new session in any meaning of session